src/System.Private.CoreLib/shared/System/DefaultBinder.cs

```diff
@@ -1204,6 +1204,63 @@ private static bool CreateParamOrder(int[] paramOrder, ParameterInfo[] pars, str
             return true;
         }
 
+        // CanChangePrimitive
+        // This will determine if the source can be converted to the target type
+        private static bool CanChangePrimitive(Type source, Type target)
```

Comment:
I think in Mono it's `internal` and used from `Array` code. Can you please check it and make it internal if that's the case?

Reply:
Indeed, thanks! I hope CoreCLR guys don't mind if use internal here

Reply:
> used from Array code. Can you please check it and make it internal if that's the case?

Is it a good idea to have a dependency from Array to DefaultBinder, TypeCode and rest of reflection? It may be a good idea to implement this using ElementTypes to make this more tree-shaking friendly.

Reply:
It's not necessarily a good idea. It was the only code that implemented primitive type widening check in managed code though (AFAIK).

Reply:
This comment still applies, but it does not prevent merging this.